### DISCUSSION TITLE
Newbie puzzled with options, variables and defaults file

### DISCUSSION THREAD

**Discussion by astesin** [mydumper/mydumper#973] [Q&A]
Dear Mr. David Ducos, first of all, thank you so much for your outstanding work! I'm not a pro DBA - but it happened so that I need a tool to migrate an extensive (approx. 150 GB) MySQL 5.5 database to a new MariaDB 10.stable DBMS, and mydumper seems to be the perfect fit for my purpose. I'm a complete newcomer, just starting to master the tool, so would you mind sharing some advice on the usage, please?

Question 1. The software is built with MySQL 5.7 libraries. Will it work well with the recent MariaDB 10 for both restore and backup? Or should I build a custom binary with MariaDB libraries instead?

Question 2. There are plenty of command line options. Which of these may be used in the defaults file and which can not? I.e. **--port 3306** option translates into a **port = 3306** directive in the file, but is it also true for every other command line option?

May I for example use the **outputdir=/home/me/backup** directive in the defaults file? Or write **events=true** in the defaults file instead of using the -E command line option?

Question 3. The **[mydumper_variables]** section of the defaults file is mentioned in the documentation. What kind of variables may be declared and initialized there? Is there a complete list of the variables and their meanings elsewhere? Can you please point me into the right direction to learn more?

Thanks in advance! Warmest regards, Andreas

**Comment by davidducos** [comment]
Hi @astesin,
Love to hear that you want to use MyDumper, for a migration you might need an expert.
Question 1: you can use both, I think that will be much easier if you use a package for your distro.
Question 2: you will need to test what command is the best for you, but for a migration is mandatory: -E -R -G  and the databases that you want -B. Depending of your case you will need to set other options. 
Question 3: usually you don't need to set session variables ([mydumper_variables]) but everything depends.

Sorry if my answer might not be what you expect, but honestly there are a lot of things that needs to be taken into account in a migration.

**Reply by astesin** [reply]
Thank you so much for your reply! Actually, my question 2 was about how the command line options syntax corresponds to their equivalents in the defaults file.
But I ended up skipping defaults files as a whole, writing shell scripts for backup and restore and resorted to command line options only this way.
Warmest regards, Andreas
